### PR TITLE
To stix

### PR DESCRIPTION
**Pull request by barvhaim**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,6 +7,7 @@
     "@carbon/colors": "^10.14.0",
     "@carbon/icons-react": "^10.19.0",
     "@carbon/layout": "^10.13.0",
+    "@carbon/motion": "^10.21.0",
```
Comment (barvhaim) [inline package.json]: remove if we don't use

File: src/components/FromSTIX/FromSTIX.jsx [hunk 2]
```diff
@@ -1,11 +1,37 @@
...
+  const stixFieldsObject = useMemo(
+    () =>
+      Object.assign(
+        ...Array.from(stixFields, (field) => ({
+          [field.type]: field.items.map((item) => item.name),
+        }))
+      ),
+    [stixFields]
+  );
+  const [coverage, officialObjectsCount, requiredObjectsCount] = getObjectsData(
```
Comment (barvhaim) [inline src/components/FromSTIX/FromSTIX.jsx]: `getObjectsData` and `coverage` naming is not clear, should it be `getStixObjectsStatistics` or something like that?

File: src/components/FromSTIX/FromSTIX.jsx [hunk 3]
```diff
@@ -16,7 +42,7 @@ const FromSTIX = () => {
 
       <div className="bx--row">
         <div className="bx--col-sm-1">
-          <AddFields />
+          <AddFields objects={coverage} />
```
Comment (barvhaim) [inline src/components/FromSTIX/FromSTIX.jsx]: it's not clear what is `coverage` objects and why is it passed as the `objects` maybe `objectsWithCoverage`? or `objects` and another prop for the coverage? `objects` are the stix objects?

File: src/components/FromSTIX/FromSTIX.jsx [hunk 4]
```diff
@@ -1,11 +1,37 @@
...
+} from "../../store/actions/from_stix";
 import Statistics from "./Statistics";
+import { getObjectsData } from "./utils";
+import { requiredFields } from "../../global/requiredFields";
 
 const FromSTIX = () => {
+  const mapping = useSelector((state) => state.fromStix.mapping);
+  const stixVersion = useSelector((state) => state.stix.stixVersion);
+  const stixFields = useSelector((state) => state.stix.stixFields);
+  const requiredSet = requiredFields[stixVersion];
```
Comment (barvhaim) [inline src/components/FromSTIX/FromSTIX.jsx]: requiredFields/requiredStixFieldsSet

File: src/components/FromSTIX/StatisticObject.jsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import styles from "./from_stix.module.scss";
+
+const StatisticObject = ({
```
Comment (barvhaim) [inline src/components/FromSTIX/StatisticObject.jsx:5]: add proptypes

File: src/components/Header/HeaderSection.jsx [hunk 6]
```diff
@@ -11,17 +13,26 @@ import {
...
       <SkipToContent />
       <HeaderName element={Link} to="/" prefix="MAP">
         My Shifter
       </HeaderName>
       <HeaderNavigation aria-label="Side navigation">
-        <HeaderMenuItem element={Link} to="/from_stix">
+        <HeaderMenuItem
+          element={Link}
+          to="/from_stix"
+          onClick={() => dispatch(changeStixVersion(STIX_VERSION.V_2_0))}
```
Comment (barvhaim) [inline src/components/Header/HeaderSection.jsx]: can't we use it as default in the reducer initial state?

File: src/components/Header/HeaderSection.jsx [hunk 7]
```diff
@@ -11,17 +13,26 @@ import {
...
+        >
           From STIX
         </HeaderMenuItem>
-        <HeaderMenuItem element={Link} to="/to_stix">
+        <HeaderMenuItem
+          element={Link}
+          to="/to_stix"
+          onClick={() => dispatch(changeStixVersion(STIX_VERSION.V_2_0))}
```
Comment (barvhaim) [inline src/components/Header/HeaderSection.jsx]: can't we use it as default in the reducer initial state?

File: src/components/STIX/AddFieldItems.jsx [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+  stixFieldId,
+  updateType,
+  objects,
+}) => {
+  const dispatch = useDispatch();
+
+  const selectStixField = (
+    value,
+    required,
+    objectName,
```
Comment (barvhaim) [inline src/components/STIX/AddFieldItems.jsx]: please check if we need those extra props

File: src/components/STIX/SelectField.jsx [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+        >
+          <FieldSearchBar />
+          <Accordion>
+            {stixFields.map((o) => (
+              <AddFieldItems
+                key={o.title}
+                title={o.title}
+                type={o.type}
+                items={o.items}
+                objectName={objectName}
```
Comment (barvhaim) [inline src/components/STIX/SelectField.jsx]: please check if we need those extra props

File: src/components/STIX/utils.js [hunk 10]
```diff
@@ -0,0 +1,13 @@
+export function filterFieldsForValue(fields, value) {
```
Comment (barvhaim) [inline src/components/STIX/utils.js:1]: need test coverage in future

File: src/components/ToSTIX/Statistics.jsx [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+  const stixTypesSet = useMemo(
+    () => new Set(Object.values(stixFields).map((field) => field.type)),
+    [stixFields]
+  );
+  const [
+    officialFieldsCount,
+    customFieldsCount,
+    officialObjectsCount,
+    customObjectsCount,
+  ] = getDataForStatistics(mapping, stixTypesSet);
```
Comment (barvhaim) [inline src/components/ToSTIX/Statistics.jsx:18]: (FYI - in "From STIX" you had different naming for that function for getting statistics)

File: src/components/ToSTIX/to_stix.module.scss [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+  }
+}
+
+.object__item_content {
+  background-color: $white-0;
+  padding: $spacing-05 $spacing-05;
+
+  .object_item__btn {
+    cursor: pointer;
+    border: 1px solid #000000;
```
Comment (barvhaim) [inline src/components/ToSTIX/to_stix.module.scss]: to do - use carbon coloring and spacing

File: src/components/ToSTIX/to_stix.module.scss [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+    @include carbon--type-style("productive-heading-03");
+    padding-bottom: $spacing-04;
+    margin-bottom: $spacing-05;
+  }
+
+  .object_item__map {
+    background-color: $white-0;
+    margin-bottom: $spacing-07;
+    padding: $spacing-05 $spacing-05;
+    border: 1.5px solid $carbon--gray-30;
```
Comment (barvhaim) [inline src/components/ToSTIX/to_stix.module.scss]: to do - use carbon spacing

File: src/components/ToSTIX/to_stix.module.scss [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+    margin-top: $spacing-05;
+    padding-left: $spacing-04;
+    padding-right: $spacing-04;
+  }
+
+  .object_item_field__header {
+    padding-top: $spacing-02;
+    padding-bottom: $spacing-02;
+    margin-bottom: $spacing-04;
+    border-bottom: 1px solid $carbon--gray-30;
```
Comment (barvhaim) [inline src/components/ToSTIX/to_stix.module.scss]: use carbon spacing

File: src/components/ToSTIX/utils.js [hunk 15]
```diff
@@ -0,0 +1,149 @@
+import { v4 as uuidv4 } from "uuid";
```
Comment (barvhaim) [inline src/components/ToSTIX/utils.js:1]: todo - test coverage in future

File: src/components/Welcome/Welcome.jsx [hunk 16]
```diff
@@ -1,13 +1,17 @@
...
 import styles from "./welcome.module.scss";
 import { useHistory } from "react-router-dom";
+import { STIX_VERSION } from "../../global/constants";
+import { changeStixVersion } from "../../store/actions/stix";
 
 const Welcome = () => {
+  const dispatch = useDispatch();
   const history = useHistory();
   const handleClickableTile = (link) => {
+    dispatch(changeStixVersion(STIX_VERSION.V_2_0));
```
Comment (barvhaim) [inline src/components/Welcome/Welcome.jsx]: move to initial state in reducer

File: src/store/actions/from_stix.js [hunk 17]
```diff
@@ -4,20 +4,20 @@ export const ADD_VALUE = "ADD_VALUE";
...
 export const UPDATE_MAPPINGS_FILTER_FIELD_VALUE =
   "UPDATE_MAPPINGS_FILTER_FIELD_VALUE";
-export const CLEAR_MAPPINGS = "CLEAR_MAPPINGS";
-export const CHANGE_STIX_VERSION = "CHANGE_STIX_VERSION";
+export const CLEAR_FROM_STIX_MAPPINGS = "CLEAR_FROM_STIX_MAPPINGS";
 export const SHOW_CUSTOM_FIELD_MODAL = "SHOW_CUSTOM_FIELD_MODAL";
 export const CLOSE_CUSTOM_FIELD_MODAL = "CLOSE_CUSTOM_FIELD_MODAL";
 
-export function addField(type, key) {
+export function addField(field, required = false) {
```
Comment (barvhaim) [inline src/store/actions/from_stix.js:15]: why do we pass the required on addField?

Comment (noaakl) [reply src/store/actions/from_stix.js:15]: for the statistics. in custom fields it's false all the time

File: src/components/STIX/AddFieldItems.jsx [hunk 18]
```diff
@@ -0,0 +1,61 @@
...
+
+const AddFieldItems = ({
+  title,
+  type,
+  items,
+  fieldNameToUpdate,
+  officialFields,
+}) => {
+  const dispatch = useDispatch();
+  const isToStix = fieldNameToUpdate ? true : false;
```
Comment (barvhaim) [inline src/components/STIX/AddFieldItems.jsx]: `!!fieldNameToUpdate`

File: src/components/STIX/AddFieldItems.jsx [hunk 19]
```diff
@@ -0,0 +1,61 @@
...
+        dispatch(closeSelectFieldModal())
+      : dispatch(addField(value, required));
+  };
+
+  return (
+    <AccordionItem
+      title={
+        isToStix
+          ? `${title}`
+          : `${title}  (${officialFields[type].size}/${items.length})`
```
Comment (barvhaim) [inline src/components/STIX/AddFieldItems.jsx]: extra spacing between title and official fields

File: src/components/STIX/AddFields.jsx [hunk 20]
```diff
@@ -1,25 +1,17 @@
...
 import AddFieldItems from "./AddFieldItems";
 import FieldSearchBar from "./FieldSearchBar";
 import ChangeVersion from "./ChangeVersion";
-import CustomField from "./CustomField";
 
-const AddFields = () => {
-  const stixFields = useSelector((state) => state.fromStix.stixFields);
+const AddFields = ({ fieldNameToUpdate, officialFields }) => {
+  const stixFields = useSelector((state) => state.stix.stixFields);
+  const isToStix = fieldNameToUpdate ? true : false;
```
Comment (barvhaim) [inline src/components/STIX/AddFields.jsx]: `!!fieldNameToUpdate`

**Pull request merged**